File: easy/maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        def bfs(tree, resultant, level):
            if tree != None:
                if len(resultant) <= level:
                    resultant.append([])
                resultant[level] = tree.val
                bfs(tree.left, resultant, level+1)
                bfs(tree.right, resultant, level+1)
        result = []
        bfs(root, result, 0)
        return len(result)

File: easy/maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
class Solution {
    int res=0;
    protected void dfs(TreeNode root, int level) {
        if (root!=null) {
            res=Integer.max(res,level);
            dfs(root.left,level+1);
            dfs(root.right,level+1);
        }
    }
    public int maxDepth(TreeNode root) {
        dfs(root,1);
        return res;
    }
}
